src/java.desktop/share/classes/javax/swing/BufferStrategyPaintManager.java

```diff
@@ -243,6 +243,13 @@ public boolean paint(JComponent paintingComponent,
                 ((SunGraphics2D)bsg).constrain(xOffset + cx, yOffset + cy,
                                                x + w, y + h);
                 bsg.setClip(x, y, w, h);
+
+                final SunGraphics2D g2d = (SunGraphics2D) bsg;
+                final Color oldBg = g2d.getBackground();
+                g2d.setBackground(paintingComponent.getBackground());
+                g2d.clearRect(x, y, w, h);
+                g2d.setBackground(oldBg);
```

Comment:
Shouldn't we limit this change to non-opaque components only? 
`if (!bufferComponent.isOpaque())`

Also, it might be worthwhile to find a way not to clear the whole screen/window area (if possible), but only the dirty part.

Reply:
Yeah, will updated to non-opaque types. Will explore on clearing the dirty regions rather than whole screen.

Reply:
I have updated to non-opaque regions and its working fine. Couldn't find a way to clear only the dirty regions.